### PR TITLE
Cleaned up the XML-RPC API implementation

### PR DESCRIPTION
**Pull request by adamretter**
This is a refactor of the XML-RPC API. Through the use of Java 8 lambda expressions we the following advantages:

1) A huge reduction in code, including the removal of duplicated code.

2) Correct use of Broker/Txn/Lock. The lambda expressions ensure that when we take a resource we always return it.

3) Correct User security.

4) Consistent exception handling.

### CODE REVIEW

**Comment by wolfgangmm** [comment]
I'll run some tests on my large test db as I did for the local implementation (batch uploads, backup/restore ...)

**Comment by adamretter** [comment]
Cool, thanks @wolfgangmm

File: src/org/exist/util/VirtualTempFile.java [hunk 1]
```diff
@@ -43,6 +43,7 @@
  * @author jmfernandez
  *
  */
+@Deprecated
 public class VirtualTempFile
```
Comment (dizzzz) [inline src/org/exist/util/VirtualTempFile.java]: Marked deprecated .... It is used on quite some locations, please could you add a reference to its replacement? (or just mention it here)

**Comment by dizzzz** [comment]
PR looks good to me

File: src/org/exist/util/VirtualTempFile.java (hunk 1 above)
Comment (adamretter) [inline src/org/exist/util/VirtualTempFile.java]: I just added some comments on that in the javadoc and rebased it for you. Hopefully we are all good now.

**Comment by wolfgangmm** [comment]
I tested this one as well, so it could be merged once the conflicts were resolved.

**Comment by adamretter** [comment]
@dizzzz that was quick! I just pushed the rebased a few seconds ago.

**Comment by dizzzz** [comment]
@adamretter sorry :-) I was just hopping by , accidentely. At your service sir :-)

**Pull request merged**